src/boot/ast.ml

```diff
@@ -124,7 +126,7 @@ and program = Program of include_ list * top list * tm
 and tm =
 | TmVar     of info * ustring * int                                 (* Variable *)
 | TmLam     of info * ustring * ty * tm                             (* Lambda abstraction *)
-| TmClos    of info * ustring * ty * tm * env                       (* Closure *)
+| TmClos    of info * ustring * ty * tm * env Lazy.t                (* Closure *)
```

Comment:
Could you say something about why `Lazy` is needed here (in the PR in general)? I'm guessing it has something to do with preventing diverting fixpoint computations?

Reply:
Conceptually, each mutually recursive lambda should become a closure with a paired environment. That environment should include the closures for all of the mutually recursive lambdas, i.e., the environment needs to be cyclic. This can be achieved in two ways: with a `fix` that creates one nesting of the cycle at a time, or an actual recursive value. OCaml allows cyclic values, but it can only handle it in very simple circumstances, e.g., `let rec x = 1 :: x`. Suffice to say that this isn't quite simple enough to support that directly, so we insert a layer of laziness, which makes it simple again.